### PR TITLE
Initial commit for odh-deployer

### PR DESCRIPTION
**Pull request by tmckayus**
The odh-deployer is moving from a subdirectory
in odh-operator-allinone to its own repository

### CODE REVIEW

**Comment by tmckayus** [comment]
@crobby ptal

**Comment by tmckayus** [comment]
just may want to drop appropriate pipeline config in here before merging

**Review by harshad16 — COMMENTED** [review]
- we need to inclued the `.aicoe-ci.yaml` file to pick up the details regarding the builds.
```
check:
  - thoth-build
build:
  build-stratergy: "Dockerfile"
  registry: "quay.io"
  registry-org: "modh"
  registry-project: "odh-deployer"
  registry-secret: "modh-pusher-secret"
```

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+CONTAINER_REPO := quay.io/opendatahub
```
Comment (harshad16) [inline Makefile:1]: if we are going to use the makefile, Shouldn't this be pointing to the `modh` quay.

**Review by harshad16 — COMMENTED** [review]
lgtm,
just a minor comment left for the makefile.

File: Makefile (hunk 1 above)
Comment (tmckayus) [reply Makefile:1]: probably, I'll take a look.  I copied the contents from odh-operator-allinone and didn't review, thanks for looking!

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
+
+ENV HOME /root
```
Comment (vpavlin) [inline Dockerfile:3]: Does this assume the container will run as root? (and the use of `/` later in general?)

Can we make it more "OpenShift SCC" friendly?

Comment (tmckayus) [reply Dockerfile:3]: let's ping @crobby  and take up in another PR. This was a migration of existing sources from odh-operator-allinone

**Pull request merged**